lib/src/types/text_search.dart

```diff
@@ -116,4 +118,255 @@ class TsVectorType extends Type<TsVector> {
   }
 }
 
-class TsQuery {}
+/// The `tsquery` type represents a search query for text search.
+///
+/// https://www.postgresql.org/docs/current/datatype-textsearch.html
+/// https://www.postgresql.org/docs/current/textsearch.html
+abstract class TsQuery {
+  int get _itemCount;
+  void _write(PgByteDataWriter writer);
+
+  static TsQuery lexeme(String text) => _LexemeTsQuery._(text, 0, 0);
+  static TsQuery not(TsQuery query) => _NotTsQuery(query);
+
+  TsQuery or(TsQuery other) => _OrTsQuery._()
+    .._add(this)
+    .._add(other);
+
+  TsQuery and(TsQuery other) => _AndTsQuery._()
```

Comment:
Personally I like using user-defined `&` and `|` operators for this. But maybe it's also better to keep consistency since we can't override `!` yet.